test/file.jl

```diff
@@ -1181,3 +1181,61 @@ if !Sys.iswindows()
         test_22566()
     end
 end  # !Sys.iswindows
+
+function test_22922()
+    def_prefix = "jl_"
+    tst_prefix = "ABCDEF"
+    mktempdir() do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, def_prefix)
+    end
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, tst_prefix)
+    end
+    # Special character prefix tests
+    tst_prefix="#!@%^&()"
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, tst_prefix)
+    end
+
+    # Behavioral differences across OS types
+    if Sys.iswindows()
+        @test_throws Base.UVError mktempdir(; prefix="*")
+        @test_throws Base.UVError mktempdir(; prefix="cdcdccd/")
+
+        # The API accepts "c:/ and c:\\" as valid prefixes.
+        # The parent directory is ignored in that case.
+
+        mkdir("c:/mydir")
+        # The temp directory created is of form "C:\\XXXXXX"
+        mktempdir("c:/mydir"; prefix="c:/") do tmpdir
+            filename = basename(tmpdir)
+            @test length(filename) == 6
+        end
+        mktempdir("c:\\mydir"; prefix="c:\\") do tmpdir
+            filename = basename(tmpdir)
+            @test length(filename) == 6
+        end
+
+    else
+        # '/' is accepted in a prefix but affects the overall path and permissions.
```

Comment:
`/` does *not* seem to be accepted, according to the test.

Reply:
`/` only as the first character of a prefix creates problems. If a path is formed by `$parent/someval/` where a path may exist in the system like say /tmp/. It will work. So technically, '/' is supported as a prefix character if the final path is writable by the process. 

"""
julia> mktempdir("/";prefix="tmp/")
"/tmp/9dLil5"

julia> 

julia> mktempdir("/";prefix="tmp/s")
"/tmp/sZZDLWl"

julia> mktempdir(;prefix="/")
ERROR: mktempdir: permission denied (EACCES)
Stacktrace:
 [1] uv_error at ./libuv.jl:68 [inlined]
 [2] #mktempdir#12(::String, ::Function, ::String) at ./file.jl:359
 [3] (::getfield(Base.Filesystem, Symbol("#kw##mktempdir")))(::Array{Any,1}, ::typeof(mktempdir), ::String) at ./<missing>:0 (repeats 2 times)

julia> mktempdir(;prefix="ss/")
ERROR: mktempdir: no such file or directory (ENOENT)
Stacktrace:
 [1] uv_error at ./libuv.jl:68 [inlined]
 [2] #mktempdir#12(::String, ::Function, ::String) at ./file.jl:359
 [3] (::getfield(Base.Filesystem, Symbol("#kw##mktempdir")))(::Array{Any,1}, ::typeof(mktempdir), ::String) at ./<missing>:0 (repeats 2 times)

julia> mkdir("/tmp/ss");mktempdir(;prefix="ss/")
"/tmp/ss/ZDAbPm"

julia> 
"""

Reply:
OK, so just make the comment more explicit.